### DISCUSSION TITLE
ru/guides/object-counting/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#10151] [Docs]
# ru/guides/object-counting/

Подсчет объектов с помощью Ultralytics YOLOv8

https://docs.ultralytics.com/ru/guides/object-counting/

**Comment by Alexts166** [comment]
Hi! Do I understand correctly that it is possible to count detected objects using a "line intersection" or "polygon" only with models that have a tracker_id in them. Such as YOLOv8 or YOLOv8-seg, but not YOLOv8-obb?

**Reply by glenn-jocher** [reply]
Hello! Yes, you're on the right track 🚀. Object counting using "line intersection" or "polygon" methods for counting relies on tracking object movement across these defined areas. For accurate counting, models like YOLOv8 and YOLOv8-seg, which provide `tracker_id` information, are indeed better suited for this task. The tracker IDs help in maintaining the identity of objects across frames, crucial for counting.

While YOLOv8-obb is excellent for oriented bounding box detections, it doesn't inherently include tracking capability, which is essential for object counting in the context of moving regions or lines.

For example, using YOLOv8n with the object counter, you might set up counting like this for a line across the frame:
```python
from ultralytics import YOLO
from ultralytics.solutions import object_counter
import cv2

model = YOLO("yolov8n.pt")
video_path = "path/to/video.mp4"
cap = cv2.VideoCapture(video_path)

line_points = [(20, 400), (1080, 400)]  # Horizontal line across the frame

counter = object_counter.ObjectCounter()
counter.set_args(view_img=True, reg_pts=line_points, draw_tracks=True)

while cap.isOpened():
    success, frame = cap.read()
    if not success: break
    tracks = model.track(frame, persist=True)
    frame = counter.start_counting(frame, tracks)
    # Optionally display or process `frame` further here
    
cap.release()
cv2.destroyAllWindows()
```
Remember, the `tracker_id` in tracks helps distinguish between different objects, crucial for accurate counts. 😊

**Reply by Alexts166** [reply]
Thank you for the answer! But still, with a large crowding and a small size of objects, in my case, the YOLOv8 and YOLOv8-seg models perform worse than YOLOv8n-obb. OK! I will look for options. As the sages say - No pain, no gain!))

**Reply by glenn-jocher** [reply]
Hey there! 👋 Thanks so much for your valuable feedback. It's interesting to hear that you're getting better performance with YOLOv8n-obb, especially in crowded scenes with smaller objects. This insight is super valuable for the community and for us to understand different use cases better!

While YOLOv8n-obb doesn't inherently include a tracker, in scenarios like yours where detection performance is key, you're on the right path in exploring options to complement its strong detection capabilities with external tracking mechanisms or solutions.

Remember, every situation has unique challenges, and it's all about finding that sweet solution that ticks all the boxes for you. "No pain, no gain!" is spot-on – keep experimenting and tweaking, and you're bound to hit that sweet spot. If you come across any interesting solutions or need further assistance, don't hesitate to share! 🚀
